### ISSUE TITLE
Dart container wont build - unable to locate packages manpages-posix and manpages-posix-dev 

### ISSUE THREAD

**Issue by mwinters-stuff** [microsoft/vscode-dev-containers#884]
- VSCode Version: 1.56.2
- Local OS Version: Ubuntu 21.04
- Name of Dev Container Definition with Issue: Dart

Steps to Reproduce:

1. Create empty folder
2. Open in VSCode
3. Add Development Container - Dart
4. Open in container
5. Build fails, unable to locate packages manpages-posix and manpages-posix-dev

These files are listed in containers/dart/.devcontainer/library-scripts/common-debian.sh

Log file:
[remoteContainers-2021-06-06T23-27-18.418Z.log](https://github.com/microsoft/vscode-dev-containers/files/6605125/remoteContainers-2021-06-06T23-27-18.418Z.log)

I will see if I can do a PR

**Comment by Chuxel** [comment]
These packages are needed for other distributions so we will need to detect and adapt.

**Comment by Chuxel** [comment]
Hmm this does not repro for me.  Do you have an old version of this definition?  There was a period where there was a bug but that should be resolved.  There should be some sed statements in `library-scripts/common-debian.sh` that reference `deb|httpredir`rather than just `deb`.  Do you see that?  If not, grab the latest script [from here](https://github.com/microsoft/vscode-dev-containers/blob/main/containers/dart/.devcontainer/library-scripts/common-debian.sh) and retry to see if that fixes it for you.

```
    # Needed for adding manpages-posix and manpages-posix-dev which are non-free packages in Debian
    if [ "${ADD_NON_FREE_PACKAGES}" = "true" ]; then
        CODENAME="$(cat /etc/os-release | grep -oE '^VERSION_CODENAME=.+$' | cut -d'=' -f2)"
        sed -i -E "s/deb http:\/\/(deb|httpredir)\.debian\.org\/debian ${CODENAME} main/deb http:\/\/\1\.debian\.org\/debian ${CODENAME} main contrib non-free/" /etc/apt/sources.list
        sed -i -E "s/deb-src http:\/\/(deb|httredir)\.debian\.org\/debian ${CODENAME} main/deb http:\/\/\1\.debian\.org\/debian ${CODENAME} main contrib non-free/" /etc/apt/sources.list
        sed -i -E "s/deb http:\/\/(deb|httpredir)\.debian\.org\/debian ${CODENAME}-updates main/deb http:\/\/\1\.debian\.org\/debian ${CODENAME}-updates main contrib non-free/" /etc/apt/sources.list
        sed -i -E "s/deb-src http:\/\/(deb|httpredir)\.debian\.org\/debian ${CODENAME}-updates main/deb http:\/\/\1\.debian\.org\/debian ${CODENAME}-updates main contrib non-free/" /etc/apt/sources.list
        sed -i "s/deb http:\/\/security\.debian\.org\/debian-security ${CODENAME}\/updates main/deb http:\/\/security\.debian\.org\/debian-security ${CODENAME}\/updates main contrib non-free/" /etc/apt/sources.list
        sed -i "s/deb-src http:\/\/security\.debian\.org\/debian-security ${CODENAME}\/updates main/deb http:\/\/security\.debian\.org\/debian-security ${CODENAME}\/updates main contrib non-free/" /etc/apt/sources.list
        sed -i "s/deb http:\/\/deb\.debian\.org\/debian ${CODENAME}-backports main/deb http:\/\/deb\.debian\.org\/debian ${CODENAME}-backports main contrib non-free/" /etc/apt/sources.list 
        sed -i "s/deb-src http:\/\/deb\.debian\.org\/debian ${CODENAME}-backports main/deb http:\/\/deb\.debian\.org\/debian ${CODENAME}-backports main contrib non-free/" /etc/apt/sources.list
        echo "Running apt-get update..."
        apt-get update
        PACKAGE_LIST="${PACKAGE_LIST} manpages-posix manpages-posix-dev"
    else
        apt-get-update-if-needed
    fi
```

**Comment by mwinters-stuff** [comment]
Hi.
This is not old - only started trying devcontainers a couple of days ago. I didn't notice that section.
I can see where it goes wrong - the ADD_NON_FREE_PACKAGES defaults to false, and if that is false, then those packages will still attempt to install.
The default docker file does not set ADD_NON_FREE_PACKAGES to true either.

**Comment by mwinters-stuff** [comment]
I have just tried on a 2nd computer (ubuntu 20.04 LTS), same result.

**Comment by Chuxel** [comment]
TLDR; Yeah it looks like the `common-debian.sh` bug I mentioned is in stable (1.56), but not insiders (1.57) which will release the next couple days.  So that may be why you are hitting this. To fix, just grab the latest version of common-debian.sh [from here](https://github.com/microsoft/vscode-dev-containers/blob/main/containers/dart/.devcontainer/library-scripts/common-debian.sh) and replace it.  This is fixed, but not yet in stable.

----

More info - See https://github.com/microsoft/vscode-dev-containers/pull/866 for the fix PR. The Dart image is weird and uses the deprecated `httpredir.debian.org` source list which now is just a redirect to `deb.debian.org`. See https://github.com/GoogleContainerTools/base-images-docker/pull/6567 for that part. Everything else in this repo does not have that problem. But, we pulled in the fix that allows this to work too.

Example line for the `common-debian.sh` version with bug:
```
sed -i "s/deb http:\/\/deb\.debian\.org\/debian ${CODENAME} main/deb http:\/\/deb\.debian\.org\/debian ${CODENAME} main contrib non-free/" /etc/apt/sources.list

```
Same line in fixed version:
```
sed -i -E "s/deb http:\/\/(deb|httpredir)\.debian\.org\/debian ${CODENAME} main/deb http:\/\/\1\.debian\.org\/debian ${CODENAME} main contrib non-free/" /etc/apt/sources.list
```

I thought this made it into a point release but it does not appear to have gotten in.

On the packages, there's actually four here:

```
manpages
manpages-dev
manapage-posix
manpages-poxsix-dev
```

The first two should be available anywhere - those are the ones that are always install.  The latter two (posix) are in the "non-free" repo for debian which is not there by default in many images.  That said, w/o add non-free being true, those are skipped.

> The default docker file does not set ADD_NON_FREE_PACKAGES to true either.

It's a bit obscured, but it is set. You can see this [here](https://github.com/microsoft/vscode-dev-containers/blob/1f920a9c4bef8596a8af32afa045b0fd8c190eb7/containers/dart/.devcontainer/Dockerfile#L15) and in your log file. By default the arguments end up with the following:

```bash
common-debian.sh "true" "vscode" "1000" "1000" "false" "true" "true"
```

The 7th arg is the one that enables this.

**Comment by mwinters-stuff** [comment]
Thanks - that PR is making it work now :) Will close this and my dodgy PR
